src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -50,37 +49,44 @@ public sealed class Column
             /// Describes how an input column should be mapped to an <see cref="IDataView"/> column.
             /// </summary>
             /// <param name="name">Name of the column.</param>
-            /// <param name="type"><see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.</param>
+            /// <param name="dataKind"><see cref="DataKind"/> of the items in the column.</param>
             /// <param name="index">Index of the column.</param>
-            public Column(string name, DataKind? type, int index)
-               : this(name, type, new[] { new Range(index) }) { }
+            public Column(string name, DataKind dataKind, int index)
+                : this(name, dataKind.ToDataKind(), new[] { new Range(index) })
+            {
+            }
 
             /// <summary>
             /// Describes how an input column should be mapped to an <see cref="IDataView"/> column.
             /// </summary>
             /// <param name="name">Name of the column.</param>
-            /// <param name="type"><see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.</param>
+            /// <param name="dataKind"><see cref="DataKind"/> of the items in the column.</param>
             /// <param name="minIndex">The minimum inclusive index of the column.</param>
             /// <param name="maxIndex">The maximum-inclusive index of the column.</param>
-            public Column(string name, DataKind? type, int minIndex, int maxIndex)
-                : this(name, type, new[] { new Range(minIndex, maxIndex) })
+            public Column(string name, DataKind dataKind, int minIndex, int maxIndex)
+                : this(name, dataKind.ToDataKind(), new[] { new Range(minIndex, maxIndex) })
+            {
+            }
+
+            public Column(string name, DataKind dataKind, Range[] source, KeyCount keyCount = null)
+                : this(name, dataKind.ToDataKind(), source, keyCount)
             {
             }
 
             /// <summary>
             /// Describes how an input column should be mapped to an <see cref="IDataView"/> column.
             /// </summary>
             /// <param name="name">Name of the column.</param>
-            /// <param name="type"><see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.</param>
+            /// <param name="kind"><see cref="InternalDataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.</param>
             /// <param name="source">Source index range(s) of the column.</param>
             /// <param name="keyCount">For a key column, this defines the range of values.</param>
-            public Column(string name, DataKind? type, Range[] source, KeyCount keyCount = null)
+            private Column(string name, InternalDataKind? kind, Range[] source, KeyCount keyCount = null)
```

Comment:
Who calls this constructor?

Reply:
Only the three public constructors above.

---
In reply to: [259520279](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259520279) [](ancestors = 259520279)